src/main/java/org/eclipse/uprotocol/uuid/factory/UUIDUtils.java

```diff
@@ -21,163 +21,191 @@
 
         private final int value;
 
-        Version(int value) {
+        private Version(int value) {
             this.value = value;
         }
 
         public int getValue() {
             return this.value;
         }
 
-        public static Version getVersion(int value) {
-            Version result = VERSION_UNKNOWN;
+        public static Optional<Version> getVersion(int value) {
             for (Version version : Version.values()) {
                 if (version.getValue() == value) {
-                    result = version;
-                    break;
+                    return Optional.of(version);
                 }
             }
-            return result;
+            return Optional.empty();
         }
     }
 
-    private UUIDUtils() {}
 
     /**
      * Convert the UUID to a String
      * @return String representation of the UUID
      */
-    public static String toString(UUID uuid) {
-        return uuid.toString();
+    public static Optional<String> toString(UUID uuid) {
+        return uuid == null ? Optional.empty() : Optional.of(uuid.toString());
     }
 
     /**
      * Convert the UUID to byte array
      * @return The byte array
      */
-    public static byte[] toBytes(UUID uuid) {
+    public static Optional<byte[]> toBytes(UUID uuid) {
+        if (uuid == null) {
+            return Optional.empty();
+        }
         byte[] b = new byte[16];
-        return ByteBuffer.wrap(b).order(ByteOrder.BIG_ENDIAN).putLong(uuid.getMostSignificantBits())
-                .putLong(uuid.getLeastSignificantBits()).array();
+        return Optional.of(ByteBuffer.wrap(b).order(ByteOrder.BIG_ENDIAN).putLong(uuid.getMostSignificantBits())
+                .putLong(uuid.getLeastSignificantBits()).array());
     }
 
     /**
      * Convert the byte array to a UUID
      * @param bytes The UUID in bytes format
      * @return  UUIDv8 object built from the byte array
      */
-    public static UUID fromBytes(byte[] bytes) {
+    public static Optional<UUID> fromBytes(byte[] bytes) {
+        if (bytes == null || bytes.length != 16) {
+            return Optional.empty();
+        }
+
         final ByteBuffer byteBuffer = ByteBuffer.wrap(bytes);
-        return new UUID(byteBuffer.getLong(), byteBuffer.getLong());
+        return Optional.of(new UUID(byteBuffer.getLong(), byteBuffer.getLong()));
     }
 
+
     /**
      * Create a UUID from the passed string
      * @param string the string representation of the uuid
-     * @return The UUIDv8 representation of the string
+     * @return The UUID object representation of the string
      */
-    public static UUID fromString(String string) {
-        return UUID.fromString(string);
+    public static Optional<UUID> fromString(String string) {
+        UUID uuid = null;
+        if (string == null || string.isBlank()) {
+            return Optional.empty();
+        }
+        try {
+            uuid = UUID.fromString(string);
+        } catch (IllegalArgumentException e) {
```

Comment:
you are swallowing exceptions - how will we know how to fix the problem - the problem should be at least logged